### PR TITLE
Fix/memory leaks

### PR DESCRIPTION
**Pull request by raphrel**

### CODE REVIEW

File: Sources/Core/State/PlayingState.swift [hunk 1]
```diff
@@ -128,34 +130,35 @@ final class PlayingState: PlayerState {
...
         }
         itemPlaybackObservingService.onFailedToPlayToEndTime = { [weak self] in self?.redirectToWaitingForNetworkState()
         }
-        itemPlaybackObservingService.onPlayToEndTime = { [weak self, context] in
-            context.delegate?.playerContext(didItemPlayToEndTime: context.currentTime)
-            context.plugins.forEach { $0.didItemPlayToEndTime(media: media, endTime: context.currentTime) }
-            if context.loopMode {
-                self?.seek(position: 0)
+        itemPlaybackObservingService.onPlayToEndTime = { [weak self] in
+            guard let strongSelf = self else { return }
```
Comment (JiceDessaint) [inline Sources/Core/State/PlayingState.swift]: You can now write:
```swift 
guard let self = self else { return }
```
I believe it is nicer

File: Example/ViewController.swift [hunk 2]
```diff
@@ -343,24 +299,25 @@ extension ViewController {
                 return duration != nil
             }
             .map { return ($0.0.0, $0.1!) } //unwrap current time & duration
-            .map(setSliderPosition)
+            .map({ [unowned self] position in
```
Comment (JiceDessaint) [inline Example/ViewController.swift]: I believe you should keep weak reference here.<br/>
Even if it is unlikely in your configuration, this closure could potentially be called with released viewController

Comment (raphrel) [reply Example/ViewController.swift]: hum, if viewController is released, disposebag is called just before no ?

File: Sources/Core/State/PlayingState.swift (hunk 1 above)
Comment (raphrel) [reply Sources/Core/State/PlayingState.swift]: not for me ;)

**Pull request merged**